### PR TITLE
Upgrade WSL 1 to 2

### LINKED ISSUE #98: Require WSL 2
WSL 1 and Ubuntu 22.04 are incompatible.  This is from @jamesjuett :

The intractable VS code debugging issue with "failed to set controlling terminal" is solved (sort of)!  The problem manifests when students have WSL 1 rather than WSL 2. There's an incompatibility with newer versions of Ubuntu (I think - I'm assuming that's why we're starting to see it this term but haven't before). Some details here: https://github.com/microsoft/WSL/issues/8356

The fix is to upgrade to WSL 2. There are two ways to do this.
- Their Windows version must be new enough. Windows 11 or Windows 10 Build 19041 or newer.
- Need to have virtualization enabled. Might need to do this in BIOS. (Although I see that less and less these days.)
- Need to have Windows Virtual Machine Platform enabled.
- May need to do step 4 here: https://learn.microsoft.com/en-us/windows/wsl/install-manual
- Finally set default version to WSL 2: https://learn.microsoft.com/en-us/windows/wsl/install#upgrade-version-from-wsl-1-to-wsl-2

We should add a sanity check and require this.

### PR DESCRIPTION
**Pull request by ohjuny**
Closes #98 
Started off a draft of a page about upgrading WSL.
Instructions taken from:
- [Issue 98](https://github.com/eecs280staff/tutorials/issues/98)
- [Microsoft 1](https://learn.microsoft.com/en-us/windows/wsl/install-manual)
- [Microsoft 2](https://learn.microsoft.com/en-us/windows/wsl/install#upgrade-version-from-wsl-1-to-wsl-2)
- [Random blog](https://www.simplilearn.com/enable-virtualization-windows-10-article)

There were a few things I was unsure about, which I have left as TODOs in the codebase.

I also do not have a Windows machine, so these instructions have not been tested.

### CODE REVIEW

**Comment by awdeorio** [comment]
I hate to ask this question since you've already put time into this, but would it be reasonable to point students at the appropriate Microsoft instructions instead of making a separate tutorial for this niche case?

**Comment by ohjuny** [comment]
> I hate to ask this question since you've already put time into this, but would it be reasonable to point students at the appropriate Microsoft instructions instead of making a separate tutorial for this niche case?

Honestly not a bad idea. I did feel like I was rewording/simplifying other tutorials while I was writing this. There is also no need for us to take on the burden of maintaining this (rather than let Microsoft take that burden) if there are no EECS 280-specific steps (which I don't think there is).

TLDR; agreed :(

**Comment by awdeorio** [comment]
Are you planning to make a separate PR with a link to the MS docs?

**Comment by ohjuny** [comment]
Yes, I will go ahead and get that set up
